### PR TITLE
API-998: Be able to purge read product table

### PR DESCRIPTION
**Pull request by willy-ahva**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Cli/PurgeReadProductTableCommand.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            'Purge table where products read through the API are stored. Keeps data for the 8 last days.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $before = new \DateTime('now - 7 days', new \DateTimeZone('UTC'));
```
Comment (fitn) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Cli/PurgeReadProductTableCommand.php]: We need data for the last 8 days + 11h for timezone. What do you think about keeping the last 10 days?

File: src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Query/DbalPurgeReadProductTableIntegration.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        $now = new \DateTime('now', new \DateTimeZone('UTC'));
+        $lastWeek = new \DateTime('now - 7 day', new \DateTimeZone('UTC'));
+        $tenBefore = new \DateTime('now - 10 day', new \DateTimeZone('UTC'));
+
+        $insertQuery = <<<SQL
+INSERT INTO akeneo_connectivity_connection_read_product (product_id, username, event_datetime)
+VALUES
+    (1, 'magento', :now),
+    (3, 'magento', :tenBefore),
```
Comment (fitn) [inline src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Query/DbalPurgeReadProductTableIntegration.php]: I think you want to use lastWeek datetime here, does not it?

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Persistence/Dbal/Query/DbalPurgeReadProductTableQuery.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function execute(\DateTime $before): int
+    {
+        $deleteQuery = <<<SQL
+DELETE FROM akeneo_connectivity_connection_read_product
```
Comment (fitn) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Persistence/Dbal/Query/DbalPurgeReadProductTableQuery.php]: The table name will change according to my comment in your previous PR

File: src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Query/DbalPurgeReadProductTableIntegration.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        $this->connection->executeUpdate(
+            $insertQuery,
+            ['now' => $now->format('Y-m-d H:i:s'), 'tenBefore' => $tenBefore->format('Y-m-d H:i:s')]
+        );
+
+        $this->purgeQuery->execute($lastWeek);
+
+        $selectQuery = <<<SQL
+SELECT product_id, username, event_datetime
+FROM akeneo_connectivity_connection_read_product
```
Comment (fitn) [inline src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Query/DbalPurgeReadProductTableIntegration.php]: Table name to change according my comment in your previous PR

File: src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Query/DbalPurgeReadProductTableIntegration.php (hunk 2 above)
Comment (willy-ahva) [reply src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Query/DbalPurgeReadProductTableIntegration.php]: nope, I want in DB date 1 | date 10. And then I removed all rows higher than 7, so the 10th one.

**Comment by willy-ahva** [comment]
> Could be done in a different workflow but you also need to:
> 
> * update docs.akeneo.com according this new crontask.
> * adds this crontask in EE repository for SaaS
> * adds this crontask on https://github.com/akeneo/cloud-paas

https://github.com/akeneo/pim-docs/pull/1397
https://github.com/akeneo/pim-enterprise-dev/pull/8150

**Pull request merged**